lib/charms/opensearch/v0/opensearch_relation_peer_cluster.py

```diff
@@ -347,7 +347,7 @@ def _rel_err_data(  # noqa: C901
         elif deployment_desc.typ == DeploymentType.OTHER:
             should_sever_relation = True
             blocked_msg = "Related to non 'main/failover'-orchestrator cluster."
-        elif orchestrators.failover_app and orchestrators.failover_app != self.charm.app.name:
+        elif orchestrators.failover_app and orchestrators.failover_app == self.charm.app.name:
```

Comment:
I believe what we had was correct, though incomplete. Perhaps: 
```suggestion
        elif orchestrators.main_app != self.charm.app.name and orchestrator.failover_app and orchestrators.failover_app != self.charm.app.name:
```